kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import {useState} from "react";
+import * as _ from "lodash";
+import "./ExpressionContainer.css"
+import {useBoxedExpressionEditorI18n} from "../../i18n";
+import {
+  Dropdown,
+  DropdownItem,
+  KebabToggle,
+  Popover,
+  SimpleList,
+  SimpleListItem,
+  SimpleListItemProps
+} from "@patternfly/react-core";
+
+export interface ExpressionContainerProps {
+  /** The name of the expression */
+  name: string,
+  /** The type of the expression */
+  type?: string,
+  /** Selected expression is already present */
+  selectedExpression?: string
+}
+
+const ExpressionContainer: (props: ExpressionContainerProps) => JSX.Element = (props: ExpressionContainerProps) => {
+  const {i18n} = useBoxedExpressionEditorI18n();
+
+  const [logicTypeIsPresent, isLogicTypeSet] = useState(!_.isEmpty(props.selectedExpression));
+  const [actionDropdownIsOpen, isActionDropdownOpen] = useState(false);
+
+  const hideSelectorMenuPopover = () => {
+    const elem = document.querySelector('.selector-menu .pf-c-button');
```

Comment:
I believe that instead of handling DOM elements, we need to change our mindset a bit. In this case we could create a small React component with a `onClick` action.

Reply:
Brief explanation of what I am doing there.
It seems that there is no way to programmatically close a Popover (https://patternfly-react.surge.sh/components/popover/) - at least, I wasn't able to find it. I was wishing that `isVisible` property could help us by changing it, but it seems not.

I am selecting the close button of the popover itself (that it is `display: none` in our case) and clicking it.
I don't know if there is a way to create a small React component around the specific content of a Patternfly component.